src/shared/extension.ts

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+/* eslint-disable no-prototype-builtins */ // Only using prototype on `Array` `Object` which are safe.
+/* eslint-disable @typescript-eslint/no-explicit-any */ // Unable to express certain generic extensions.
+
+export {};
+
+// Causing colorization issues if placed above Array.prototype...
+// Ideally: ((_) => number) | ((_) => string)
+type Selector<T> = (_: T) => number | string
+
+declare global {
+    interface ArrayConstructor {
+        commonLength(a: any[], b: any[]): number
+    }
+    interface Array<T> {
+        last: T
+        replace(items: T[]): void // From Mobx, but not showing up.
+        remove(item: T): boolean // From Mobx, but not showing up.
+        removeWhere(predicate: (item: T) => boolean): T | false
+        sortBy<T>(this: T[], selector: Selector<T>, descending?: boolean): Array<T> // Not a copy
+    }
+    interface String {
+        file: string
+        path: string
+    }
+}
+
+!Array.hasOwnProperty('commonLength') &&
+Object.defineProperty(Array, 'commonLength', {
+    value: function(a: any[], b: any[]): number {
+        let i = 0;
+        // eslint-disable-next-line no-empty
+        for (; a[i] === b[i] && i < a.length && i < b.length; i++) {}
+        return i;
+    }
+});
+
+!Array.prototype.hasOwnProperty('last') &&
+Object.defineProperty(Array.prototype, 'last', {
+    get: function() {
+        return this[this.length - 1];
+    }
+});
+
+!Array.prototype.hasOwnProperty('removeWhere') &&
```

Comment:
>removeWhere [](start = 33, length = 11)

This name is interesting because it only removes the "first" item and not "all" items that match the predicate. Can we rename this?

Reply:
Renamed to `removeFirstWhere`.

---
In reply to: [441659711](https://github.com/microsoft/sarif-vscode-extension/pull/282#discussion_r441659711) [](ancestors = 441659711)